platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDbTableConfig.java

```diff
@@ -106,7 +106,7 @@ private static final class ClassVersion {
     /**
      * Max number of keys that can be stored in a table.
      */
-    private long maxNumberOfKeys = 0;
```

Comment:
Long time ago, this was just a hint used to calculate the (unchangeable) number of HDHM bucket. Then this hint was changed to a hard limit. Now it's a hint again, and I'm renaming it to "initialCapacity" (similar to HashMap) to reflect that